pipelines/youtube_data/__init__.py

```diff
@@ -0,0 +1,181 @@
+import datetime
+from typing import Any, Dict, Iterator, List, Sequence
+
+from googleapiclient.discovery import build
+
+import dlt
+from dlt.extract.source import DltResource, DltSource
+from dlt.common.typing import TDataItem
+
+
+_SERVICE_NAME = "youtube"
+_SERVICE_VERSION = "v3"
+current_time = datetime.datetime.now()
+
+def _get_channel_id(youtube: Any, channel_name: str) -> List[str]:
+    """
+    function to retrieve channel id given channel custom name e.g. MrBeast
+
+    params:
+        youtube: YouTube resource object
+        channel_name: YouTube custom channel name
+
+    return:
+        channel_id: YouTube channel id
+    """
+    response = youtube.search().list(
+        part="snippet",
+        q={channel_name},
+        type="channel"
+    ).execute()
+
+    for item in response["items"]:
+        #filter only youtube channel
+        if item["id"]["kind"] == "youtube#channel":
+            return item["id"]["channelId"]
+
+def _get_channel_video(
+        youtube: Any, 
+        channel_id: List[str], 
```

Comment:
The `channel_id` argument is typed as `List[str]` but it seems to be treated as a single string in the function call.